### PR TITLE
STORM-689. SimpleACLAuthorizer should provide a way to restrict who can submit topologies.

### PR DESCRIPTION
**Pull request by harshach**

### CODE REVIEW

**Comment by harshach** [comment]
@revans2 can you please take a look at this.

**Comment by harshach** [comment]
@revans2  pinging for a review. Thanks.

File: storm-core/src/jvm/backtype/storm/Config.java [hunk 1]
```diff
@@ -339,6 +339,13 @@
     public static final Object NIMBUS_ADMINS_SCHEMA = ConfigValidation.StringsValidator;
 
     /**
+     * A list of users that are the only ones allowed to run user operation on storm cluster.
+     * To use this set nimbus.authorizer to backtype.storm.security.auth.authorizer.SimpleACLAuthorizer
+     */
+    public static final String NIMBUS_USERS = "nimbus.users";
```
Comment (Parth-Brahmbhatt) [inline storm-core/src/jvm/backtype/storm/Config.java]: We should modify security.MD to document this config.

**Comment by Parth-Brahmbhatt** [comment]
+1. I think most users would much rather define a nimbus.groups m e can probably file a separate jira for that.

File: storm-core/src/jvm/backtype/storm/security/auth/authorizer/SimpleACLAuthorizer.java [hunk 2]
```diff
@@ -96,7 +101,10 @@ public boolean permit(ReqContext context, String operation, Map topology_conf) {
         }
 
         if (_userCommands.contains(operation)) {
-            return true;
+            if (_nimbusUsers.size() > 0 && _nimbusUsers.contains(user))
+                return true;
+            else if (_nimbusUsers.size() == 0)
+                return true;
```
Comment (revans2) [inline storm-core/src/jvm/backtype/storm/security/auth/authorizer/SimpleACLAuthorizer.java]: I personally think 
```
return _nimbusUsers.size() == 0 || _nimbusUsers.contains(user);
```
is more readable than what is here, but this is really minor.

**Comment by revans2** [comment]
Sorry it took so long for the review.  It looks good, and I agree having groups of allowed users would be a good step.

**Comment by harshach** [comment]
@Parth-Brahmbhatt @revans2 Thanks for the review. I updated the PR to include nimbus.groups as well and added doc under SECURITY.md. Please take a look.

**Comment by Parth-Brahmbhatt** [comment]
+1.

**Comment by vesense** [comment]
+1

**Comment by revans2** [comment]
+1

**Pull request merged**